torchbiggraph/examples/e2e/data_prep.py

```diff
@@ -0,0 +1,325 @@
+    nentities_postmap = 0
+    for lhs_part in range(NPARTS):
+        for rhs_part in range(NPARTS):
+            nentities_postmap += conn.execute(f"""
+            select count(*) from edges_{lhs_part}_{rhs_part}
+            """).fetchall()[0][0]
+    
+    if nentities_postmap != nentities_premap:
+        logging.warning("DROPPED EDGES DURING REMAPPING.")
+        logging.warning(f"We started with {nentities_premap} and finished with {nentities_postmap}")
+
+    end = time.time()
+    logging.info(f"Remapped edges in {end - start}s")
+
+
+def load_edges(fname, conn):
+    """
+    A simple function to load the edges into the SQL table. It is
+    assumed that we will have a file of the form:
+    | source_id | source_type | relationship_name | destination_id | destination_type |
+    """
+    logging.info("Loading edges")
+    start = time.time()
+    cur = conn.cursor()
+    cur.executescript("""
+    DROP TABLE IF EXISTS edges
+    ;
+
+    CREATE TABLE edges (
+        source_id INTEGER,
+        source_type TEXT,
+        destination_id INTEGER,
+        destination_type TEXT,
+        rel TEXT
+    )
+    """)
+
+    edges = pd.read_csv(fname)
+    edges.to_sql('edges', conn, if_exists='append', index=False)
+    end = time.time()
+    logging.info(f"Loading edges in {end - start}s")
+
+
+def write_relations(outdir, rels, conn):
+    """
+    A simple function to write the relevant relationship information out
+    for training.
+    """
+    logging.info("Writing relations for training")
+    start = time.time()
+    out = rels.sort_values('graph_id')['id'].to_list()
+    with open(f'{outdir}/dynamic_rel_names.json', mode='w') as f:
+        json.dump(out, f, indent=4)
+    end = time.time()
+    logging.info(f"Wrote relations in {end - start}s")
+
+
+def write_single_edge(work_packet):
+    """
+    A function to write out a single edge-lists in the format that
+    PyTorch BigGraph expects.
+
+    The work packet is expected to come contain information about
+    the lhs and rhs partitions for these edges, the directory
+    where we should put this information, and the database
+    connection that we should use.
+    """
+    lhs_part, rhs_part, outdir, conn = work_packet
+    query = f"""
+    select *
+    from edges_{lhs_part}_{rhs_part}
+    ;
+    """
+    df = pd.read_sql_query(query, conn)
+    print(query)
+    out_name = f'{outdir}/edges_{lhs_part}_{rhs_part}.h5'
+    with h5py.File(out_name, mode='w') as f:
+        # we need this for https://github.com/facebookresearch/PyTorch-BigGraph/blob/main/torchbiggraph/graph_storages.py#L400
+        f.attrs['format_version'] = 1
+        for dset, colname in [('lhs', 'source_id'), ('rhs', 'destination_id'), ('rel', 'rel_id')]:
+            f.create_dataset(dset, dtype='i', shape=(len(df),), maxshape=(None, ))
+            f[dset][0 : len(df)] = df[colname].tolist()
+
+
+def write_edges(outdir, LHS_PARTS, RHS_PARTS, conn):
```

Comment:
```suggestion
def write_all_buckets(outdir, lhs_parts, rhs_parts, conn):
```

We use `buckets` for (i, j) partition pairs.

Also, python typically uses ALL_UPPERCASE for constants only.